From 3015279966aed84fee81f04cd791c8458fddf4b0 Mon Sep 17 00:00:00 2001
From: David Disseldorp <ddiss@suse.de>
Date: Thu, 2 Jul 2015 11:20:47 +0200
Subject: [PATCH 062/213] ceph: remove xattr when null value is given to
 setxattr()
References: fate#318918
Git-commit: bcdfeb2eb4e42b811950b9cd226109291051732a
Patch-mainline: v3.14-rc4

For the setxattr request, introduce a new flag CEPH_XATTR_REMOVE
to distinguish null value case from the zero-length value case.

Signed-off-by: Yan, Zheng <zheng.z.yan@intel.com>
[ddiss@suse.de: pull in libceph change only: new flag definition]
Acked-by: David Disseldorp <ddiss@suse.de>

---
 include/linux/ceph/ceph_fs.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/include/linux/ceph/ceph_fs.h b/include/linux/ceph/ceph_fs.h
index 2ad7b86..31a2dce 100644
--- a/include/linux/ceph/ceph_fs.h
+++ b/include/linux/ceph/ceph_fs.h
@@ -348,8 +348,9 @@ extern const char *ceph_mds_op_name(int op);
 /*
  * Ceph setxattr request flags.
  */
-#define CEPH_XATTR_CREATE  1
-#define CEPH_XATTR_REPLACE 2
+#define CEPH_XATTR_CREATE  (1 << 0)
+#define CEPH_XATTR_REPLACE (1 << 1)
+#define CEPH_XATTR_REMOVE  (1 << 31)
 
 union ceph_mds_request_args {
 	struct {
-- 
2.1.4

